src/oracle/types/primitives.rs

```diff
@@ -0,0 +1,23 @@
+use std::error::Error;
+
+use diesel::deserialize::FromSql;
+use diesel::sql_types::*;
+
+use oracle::backend::Oracle;
+
+use super::super::connection::OracleValue;
+
+impl FromSql<Text, Oracle> for String {
+    fn from_sql(bytes: Option<&OracleValue>) -> Result<Self, Box<Error + Send + Sync>> {
+        let bytes = not_none!(bytes);
+        let bytes = bytes.bytes.clone();
```

Comment:
Why do we try to use our own fragile implementation of `CStr` here?

This could be written as:

```rust
Ok(CStr::from_bytes_with_nul(bytes)?.to_str()?.to_owned())
```